### PR TITLE
feat(financial-templates-lib): preserve url in truncated discord ticket

### PR DESCRIPTION
**Pull request by Reinis-FRP**
**Motivation**

Improve Discord Ticket verification by trying to preserve URLs in the message when truncating due to Discord character limit. This would allow to retain sufficient pointers (e.g. Oracle UI link) when parts of the message are truncated.

**Summary**

Modifies Discord Ticket message truncating logic to try preserve included URLs.

**Details**

This may be unnecessary for some PRs. Catch-all for detailed explanations about the implementation decisions and implications of the change.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [x]  Untested

**Issue(s)**

Fixes #XXXX

### CODE REVIEW

**Review by daywiss — APPROVED** [review]
seems fine, but my gut says there was probably a simpler way that could avoid altering log contents.

maybe by throwing an error if log was too long, to make it the responsibility of application logging to keep messages shorter, or by maybe adding a special discord only field, or separate logger for application to fill out in the case of logging to discord for this purpose, which again restricts lengths of logs explicitly

File: packages/financial-templates-lib/src/logger/DiscordTicketTransport.ts [hunk 1]
```diff
@@ -150,9 +156,57 @@ export class DiscordTicketTransport extends Transport {
...
+
+    // Regular expression to match URLs.
+    const urlRegex = /(https?:\/\/[^\s]+)/g;
+    const urls = message.match(urlRegex);
+
+    // If there are no URLs, just truncate the end of the message.
+    if (urls === null) return message.slice(0, limit - TRUNCATED.length) + TRUNCATED;
+
+    // Split the message into chunks around URLs for further processing.
+    const messageChunks = message.split(urlRegex);
```
Comment (mrice32) [inline packages/financial-templates-lib/src/logger/DiscordTicketTransport.ts:174]: Interesting! Didn't realize you could do this with split!

**Review by mrice32 — APPROVED** [review]
LGTM! Like @daywiss, I worry a bit about the complexity, but as long as errors are bubbled up, I think this is okay!

**Comment by Reinis-FRP** [comment]
> Looks good! A test would be nice robot

Agree, added some truncation tests in 75d280bffba9587596236898680eb1ebb430bd7d

**Pull request merged**